asdf/generic_io.py

```diff
@@ -615,6 +616,19 @@ def memmap_array(self, offset, size):
         """
         raise NotImplementedError()
 
+    def close_memmap(self):
+        """
+        Close the memmapped file (if one was mapped with memmap_array)
+        """
+        raise NotImplementedError()
+
+    def flush_memmap(self):
+        """
+        Flush any pending writes to the memmapped file (if one was mapped with
+        memmap_array)
+        """
+        raise NotImplementedError()
+
```

Comment:
Could you add an error message here describing that this should be implemented if needed.

Also, do we need these implemented by all the child classes?

Reply:
I added a rather generic error message to these (and the NotImplementedError in memmap_array). Improvements to the wording are appreciated.

I believe all of the calls to memmap_array, flush_memmap and close_memmap are within conditional blocks that check can_memmap. RealFile is the only subclass that returns True for can_memmap (I'm not sure if others would be possible) and RealFile has all of these implemented. I want to say a mixin might make sense here but I am open to suggestions.

Reply:
I dislike mixins in general. I was more asking in case all the subclasses where implementing their own version in which case we would want this to be an abc method which is required to be implemented by the subclasses rather than having the error message.